presto-docs/src/main/sphinx/plugin/redis-hbo-provider.rst

```diff
@@ -0,0 +1,61 @@
+=======================
+Redis Hbo Provider
+=======================
+
+Redis Hbo Provider allows loading a custom configured Redis Client for storing/retrieving Historical Stats. The Redis client is stateful and is based on
+Lettuce https://github.com/lettuce-io/lettuce-core
+
+Configuration
+-------------
+
+Create ``etc/catalog/hbo-provider.properties`` to mount the Redis Hbo Provider Plugin, replacing the properties as appropriate:
+
+Configuration properties
+------------------------
+
+The following configuration properties are available:
+
+===================================     =====================================================================
+Property Name                           Description
+===================================     =====================================================================
+``coordinator``                         Boolean property whether presto server a coordinator
+``hbo.provider.server_uri``             Redis Server URI
+``hbo.provider.total-fetch-timeoutms``  Maximum timeout in ms for redis fetch requests
+``hbo.provider.total-set-timeoutms``    Maximum timeout in ms for redis set requests
+``hbo.provider.default-ttl-sec``        TTL in sec of the redis data to be stored
+``hbo.provider.enabled``                Boolean property whether this plugin in enabled in production
+``credentials-path``                    Path for redis credentials
+===================================     =====================================================================
+
+Credentials
+-----------
+
+The connector requires the redis server server uri in order to access Redis ``hbo.provider.server_uri``
+Based on your custom redis client you may need to add additional credentials
+
+
+ Setup
+------------------------
+You should place the JAR file into the plugins directory, insert the required configuration where needed, and it will then be loaded dynamically during runtime.
+
+Steps
+1. Add the following to register the plugin in <fileSets> in presto-server/src/main/assembly/presto.xml
+ ``
+ <fileSet>
+             <directory>${project.build.directory}/dependency/redis-hbo-provider-${project.version}</directory>
+             <outputDirectory>plugin/redis-hbo-provider</outputDirectory>
+ </fileSet>
+ ``
+2. Add META-INF.services file in presto-hbo-provider/src/main/resources with the Plugin entry class com.facebook.presto.statistic.RedisProviderPlugin
+3. Add dependency on the module in presto-server/pom.xml
+ ``
+ <dependency>
+            <groupId>com.facebook.presto</groupId>
+            <artifactId>presto-hbo-provider</artifactId>
+            <version>${project.version}</version>
+            <type>zip</type>
+            <scope>provided</scope>
+ </dependency>
+``
+
+4. Add a custom redis client in src/main/java/com/facebook/presto/statistic/RedisClusterClientProvider.java. Please note that a regular RedisClient might be sufficient; it's not obligatory to utilize a RedisClusterClient.
```

Comment:
Why would one need a custom redis client?

Reply:
There can be multiple redis deployment strategies - [eg](https://medium.com/hepsiburadatech/redis-solutions-standalone-vs-sentinel-vs-cluster-f46e703307a9) Standalone, sentinal or cluster. 

I think lettuce supports Standalone and sentinal deployments with [RedisClient](https://github.com/lettuce-io/lettuce-core/blob/bfb6880afb4fbb70110f5cd8252af69806a104d9/src/main/java/io/lettuce/core/RedisClient.java#L89).

RedisClusterClient with [RedisClusterClient](https://github.com/lettuce-io/lettuce-core/blob/bfb6880afb4fbb70110f5cd8252af69806a104d9/src/main/java/io/lettuce/core/cluster/RedisClusterClient.java)

This implementation focuses on clusterClient.
 But on thinking more i think we can remove dependency on the client and make [RedisPlanStatisticsProvider](https://github.com/prestodb/presto/pull/20455/files#diff-1f960ef7009cb29e42148067ecc2ae4715b6f594d1ce037e27c7455e789400eb) depend on the **RedisAsyncCommands<String, HistoricalPlanStatistics>** interface instead.